### PR TITLE
Add debug logging to `autosectionlabel`

### PR DESCRIPTION
**Pull request by patzm**
Subject: I started using emoji and other non-plain-text content in section headings. I other wonder what the auto-generated section label will be.

- Feature

### Purpose
See the subject. For instance, the following heading
```rst
|:page_facing_up:| Formatting
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
gets rendered (using [`sphinxemoji`](https://sphinxemojicodes.readthedocs.io/en/stable/#)) as 
image

Trying to reference it with
```rst
:ref:`|:page_facing_up:| Formatting`
```
~fails~ worked. I guess I did something else wrong. I had no clue then whether it was due to the `|` or `:` characters, whether the emoji had already been inserted and the correct reference would have been ````:ref:`📄 Formatting` ````, or something else altogether. I also couldn't find anything in the docs 😕. Because the code internally relies on `docutils`, I didn't want to rabbit hole and think about what the code would actually do.

Output would be for instance:
```
/some/dir/docs/shared/contribution/code_authoring.rst:40: section "|:page_facing_up:| Formatting" gets labeled as "|:page_facing_up:| formatting"
/some/dir/docs/shared/contribution/code_authoring.rst:50: section "|:pencil2:| Writing new code" gets labeled as "|:pencil2:| writing new code"
/some/dir/docs/shared/contribution/documentation.rst:2: section "Documentation" gets labeled as "documentation"
```

<details><summary>wrong conclusion</summary>
What worked eventually:
```rst
:ref:`|:page_facing_up:| formatting`
```
~note the _lower-case_ `formatting`, not _upper-case_.~
</details>

### Questions
Soooo: debug logging. Is this the best way to do this? I could configure the verbosity then with `-vv`([docs](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-v)).

Should we also add something to the docs? Somehow help other lost souls like myself?

### CODE REVIEW

**Comment by AA-Turner** [comment]
Please target against 5.x and add a CHANGES entry and tests.

A

**Comment by patzm** [comment]
Done. I also added documentation in 4b31161406ae606eeffbe12a7f4fff1dc72dd2b9. Renders as 
image

Changes added ✅.

Can you help me with the tests? I only see tests that look at the generated `html` or `xml` output. But this is actually for testing whether the debug logging works. Does that even need a test?

File: doc/usage/extensions/autosectionlabel.rst [hunk 1]
```diff
@@ -45,3 +45,11 @@ Configuration
    example, when set 1 to ``autosectionlabel_maxdepth``, labels are generated
    only for top level sections, and deeper sections are not labeled.  It
    defaults to ``None`` (disabled).
+
+
+Debugging
+---------
```
Comment (AA-Turner) [inline doc/usage/extensions/autosectionlabel.rst:51]:
```suggestion
Debugging
---------

```

**Comment by AA-Turner** [comment]
> Please target against 5.x

I changed the base but you'll need to rebase to remove all the extra commits.

A

File: doc/usage/extensions/autosectionlabel.rst [hunk 2]
```diff
@@ -45,3 +45,11 @@ Configuration
...
+The ``WARNING: undefined label`` indicates that your reference in
```
Comment (AA-Turner) [inline doc/usage/extensions/autosectionlabel.rst]:
```suggestion
---------

The ``WARNING: undefined label`` indicates that your reference in
```

File: CHANGES [hunk 3]
```diff
@@ -12,6 +12,8 @@ Deprecated
 
 Features added
 --------------
+* #10881: make debugging automatically generated section labels by
+  ``sphinx.ext.autosectionlabel`` easier.
 
 * #10840: One can cross-reference including an option value: ``:option:`--module=foobar```.
   Patch by Martin Liska.
```
Comment (AA-Turner) [inline CHANGES]:
```suggestion

* #10840: One can cross-reference including an option value: ``:option:`--module=foobar```.
  Patch by Martin Liska.
* #10881: autosectionlabel: Record the generated section label to the debug log.
```

**Comment by AA-Turner** [comment]
Thanks Martin!

A

**Pull request merged**